### PR TITLE
Update java-reference-types.md

### PR DESCRIPTION
**Pull request by susiteemu**
Interesting shit! Put few comments among the text.

btw, misusing this but whatever :D

### CODE REVIEW

**Comment by Arttuv** [comment]
Thanks for the comments! I've refactored the text, what should I do now? :D

File: content/post/java-reference-types.md [hunk 1]
```diff
@@ -36,6 +39,8 @@ However, reference objects are just like any other Java objects: they are create
 
 ## Collections
 
+[TT]: Is there a collection for soft references or phantom references? If not, maybe make this a subchapter for Weak reference?
+
```
Comment (Arttuv) [inline content/post/java-reference-types.md:43]: Good idea.

File: content/post/java-reference-types.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ draft = true
 title = "Reference types in Java"
 +++
 
+[TT]: Is the purpose that the article starts here or is there going to some sort of introduction before? It might benefit from having a sort of short introduction for all the possible reference types; now the text jumps straight into weak and soft references, but later on introduces some more.
```
Comment (Arttuv) [inline content/post/java-reference-types.md:7]: True, I've added an introduction and moved the reachability chapter to top.

File: content/post/java-reference-types.md [hunk 3]
```diff
@@ -4,6 +4,9 @@ draft = true
...
+[TT]: Is there a strong reference? It is referenced (pun intented) in few places, but there is no separate chapter for it. Or is the reachability kind of the same thing?
```
Comment (Arttuv) [inline content/post/java-reference-types.md:8]: Good point, I've added explanation on strong reference as well.

**Comment by susiteemu** [comment]
1. Publish
2.??? 
3. Profit

Good changes :D, closing this "pull request"

**Pull request closed without merging**